addons/pos_mercado_pago/models/pos_payment_method.py

```diff
@@ -0,0 +1,104 @@
+
+    def force_pdv(self):
+        """
+        Triggered in debug mode when user change want to force the "PDV" mode
+        """
+        mercado_pago = MercadoPagoPosRequest(self.mp_bearer_token)
+        _logger.info('Calling Mercado Pago to force the terminal mode to "PDV"')
+        mode = {"operating_mode": "PDV"}
+        mercado_pago._call_mercado_pago("patch", f"/point/integration-api/devices/{self.mp_id_point_smart}", mode)
+        return None
+
+    def mp_payment_intent_create(self, infos):
+        """
+        Called from frontend for creating a payment intent in Mercado Pago
+        """
+        mercado_pago = MercadoPagoPosRequest(self.mp_bearer_token)
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Mercado Pago"))
+
+        # Call Mercado Pago for payment intend creation
+        resp = mercado_pago._call_mercado_pago("post", f"/point/integration-api/devices/{self.mp_id_point_smart}/payment-intents", infos)
+        _logger.info("mp_payment_intent_create(), response from Mercado Pago: %s", resp)
+        return resp
+
+    def mp_payment_intent_get(self, payment_intent_id):
+        """
+        Called from frontend to get the last payment intend from Mercado Pago
+        """
+        mercado_pago = MercadoPagoPosRequest(self.mp_bearer_token)
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Mercado Pago"))
+
+        # Call Mercado Pago for payment intend status
+        resp = mercado_pago._call_mercado_pago("get", f"/point/integration-api/payment-intents/{payment_intent_id}", {})
+        _logger.info("mp_payment_intent_get(), response from Mercado Pago: %s", resp)
+        return resp
+
+    def mp_payment_intent_cancel(self, payment_intent_id):
+        """
+        Called from frontend to cancel a payment intent in Mercado Pago
+        """
+        mercado_pago = MercadoPagoPosRequest(self.mp_bearer_token)
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Mercado Pago"))
+
+        # Call Mercado Pago for payment intend cancelation
+        resp = mercado_pago._call_mercado_pago("delete", f"/point/integration-api/devices/{self.mp_id_point_smart}/payment-intents/{payment_intent_id}", {})
+        _logger.info("mp_payment_intent_cancel(), response from Mercado Pago: %s", resp)
+        return resp
+
+    def _find_terminal(self, vals):
+        mercado_pago = MercadoPagoPosRequest(vals['mp_bearer_token'])
+        data = mercado_pago._call_mercado_pago("get", "/point/integration-api/devices", {})
+        if 'devices' in data:
+            # Search for a device id that contains the serial number entered by the user
+            found_device = None
+            for device in data['devices']:
```

Comment:
do you try a Python if Statement in one Line ?
```suggestion
            found_device = next( device for device in data['devices'] if device.get('id') in  vals['mp_id_point_smart'])
```

Reply:
Not lisible and it's two lines since everything on one line is too long